### PR TITLE
Fix glob for nested directories

### LINKED ISSUE #75: Glob errors when trashing nested directories
On macOS and 4.3.0. 

I'm trying to trash a directory with nested directories like this:
```
foo
|____qux.txt
|____bar
| |____baz.txt
```
When I run:
`trash(['foo/**', '!foo'], {glob: true})`

I get the following error:
```
Error: Command failed: /Users/lilaconlee/stuff/trash-test/node_modules/trash/lib/macos-trash /Users/lilaconlee/stuff/trash-test/foo/bar /Users/lilaconlee/stuff/trash-test/foo/bar/baz.txt /Users/lilaconlee/stuff/trash-test/foo/qux.txt

The file “baz.txt” doesn’t exist.
```

This results in most everything getting trashed appropriately, except `qux.txt.` that never gets trashed after the previous command fails.

### PR DESCRIPTION
**Pull request by medusalix**
Makes it possible to use the option `glob` with nested directories through sorting the filepaths by their depth before passing them to the `trash` function.

Fixes #75

### CODE REVIEW

**Comment by sindresorhus** [comment]
This way works fine, but I had another idea in mind. If we have `['foo', 'foo/bar', 'foo/baz']`, couldn't we just remove `foo/bar` and `foo/baz`? As `foo` will remove those anyway. I think this would help speed up the deletion process too and reduce the situations where users encounter https://github.com/sindresorhus/trash-cli/issues/9.

**Comment by medusalix** [comment]
@sindresorhus Your proposed solution is actually slower the way I have implemented it 😄. I traverse the whole array for each item to check if there's already a directory above it.

To be honest, I think the operation system is much quicker at moving files to the trash than JS is at de-duplicating an array.

**Comment by sindresorhus** [comment]
> Your proposed solution is actually slower the way I have implemented it

How was that measured? The difference is probably negligible for macOS, but for Linux we actually remove each file using JS, so I think my solution would be more efficient then.

> To be honest, I think the operation system is much quicker at moving files to the trash than JS is at de-duplicating an array.

I highly doubt that.

File: index.js [hunk 1]
```diff
@@ -25,6 +25,9 @@ module.exports = (paths, options) => pTry(() => {
 	}
 
 	paths = paths
+		.filter(filePath => !paths.some(
+			otherPath => filePath !== otherPath && path.dirname(filePath) === otherPath)
+		)
```
Comment (sindresorhus) [inline index.js]: This needs to be after the `.map` as it will only work correctly after the path is made absolute.

File: index.js [hunk 2]
```diff
@@ -25,6 +25,9 @@ module.exports = (paths, options) => pTry(() => {
 	}
 
 	paths = paths
+		.filter(filePath => !paths.some(
+			otherPath => filePath !== otherPath && path.dirname(filePath) === otherPath)
```
Comment (sindresorhus) [inline index.js]: Use https://github.com/sindresorhus/is-path-inside

**Pull request merged**